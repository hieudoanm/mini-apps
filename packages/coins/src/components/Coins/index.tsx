import React from 'react';
import { Coin } from '../../types';
import { currencyFormatter } from '../../utils/currency-formatter';
import List from '../List';

const Coins: React.FC<{ coins: Coin[] }> = ({ coins = [] }) => {
  return (
    <>
      {coins.map((coin: Coin) => {
        const { uuid, name, price, iconUrl, coinrankingUrl, marketCap } = coin;
        return (
          <List.Item key={`coin-${uuid}`}>
            <div className="flex items-center justify-between">
              <div className="flex items-center gap-4">
                <div
                  className="w-8 h-8 bg-cover bg-center"
                  style={{ backgroundImage: `url(${iconUrl})` }}
                ></div>
                <a href={coinrankingUrl} target="_blank" rel="noreferrer">
                  {name}
                </a>
              </div>
              <div>
                <b>{currencyFormatter(parseFloat(price))}</b>
                <span className="ml-1">
                  ({currencyFormatter(parseFloat(marketCap))})
                </span>
              </div>
            </div>
          </List.Item>
        );
      })}
    </>
  );
};

export default Coins;
