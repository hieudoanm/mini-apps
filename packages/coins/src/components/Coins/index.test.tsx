import { render } from '@testing-library/react';
import Coins from '.';

describe('Coins', () => {
  test('render default', () => {
    const wrapper = render(<Coins coins={[]} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
