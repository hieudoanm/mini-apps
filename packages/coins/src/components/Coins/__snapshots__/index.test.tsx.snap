// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins render default 1`] = `<div />`;
